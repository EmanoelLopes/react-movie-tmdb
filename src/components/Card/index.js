import React from 'react';
import styled from 'styled-components';
import { string } from 'prop-types';
import { Link } from 'react-router-dom';

const CardFeatured = styled.div`
  border: 1px solid #ddd;
  box-shadow: 2px 2px 5px #ddd;
  display: flex;

  .card-poster figure,
  .card-poster img {
    height: 100%;
  }

  .card-poster img {
    display: block;
    width: 185px;
  }

  .card-info {
    padding: 15px;
  }

  .card-info h3 {
    font-size: 22px;
    margin-bottom: 15px;
  }
`;

const Card = props => {
  const { title, description, poster, id } = props;

  return (
    <CardFeatured>
      <div className="card-poster">
        <figure>
          <img src={`http://image.tmdb.org/t/p/w500${poster}`} alt={title}/>
        </figure>
      </div>
      <div className="card-info">
        <h3>{title}</h3>
        <p>{description}</p>
        <Link to={id}>More info</Link>
      </div>
    </CardFeatured>
  );
};

Card.propTypes = {
  poster: string,
  title: string,
  description: string,
  id: string,
};

Card.defaultProps = {
  poster: '',
  title: '',
  description: '',
  id: '',
};

export default Card;
