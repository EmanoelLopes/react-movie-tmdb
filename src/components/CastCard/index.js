import React from 'react';
import { number, string } from 'prop-types';
import { images } from 'utils/styled';
import { CastContainer } from './styles';

const CastCard = props => {
  const {
    id,
    profile,
    name,
    character
  } = props;

  return (
    <CastContainer key={id}>
      <img
        src={(profile)
            ? `${images.path}${images.profile}${profile}`
            : `${images.placeholder.profile}`
        }
        alt={name}
      />
      <h4>{`${name} (${character})`}</h4>
    </CastContainer>
  );
};

CastCard.propTypes = {
  id: number.isRequired,
  profile: string,
  name: string.isRequired,
  character: string.isRequired,
};

CastCard.defaultProps = {
  profile: `${images.placeholder.profile}`,
};

export default CastCard;