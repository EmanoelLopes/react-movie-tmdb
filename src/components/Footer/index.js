import React from 'react';

function Footer () {
  return (
    <footer className="rm-footer">
      <div className="container">
        Copyright &copy; 2019
      </div>
    </footer>
  );
}

export default Footer;
