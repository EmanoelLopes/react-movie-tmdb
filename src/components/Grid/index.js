import React from 'react';
import styled from 'styled-components';
import { node } from 'prop-types';

const GridContainer = styled.div`
  display: grid;
  grid-template-columns: 50% 50%;
  grid-template-rows: auto; 
  grid-column-gap: 10px;
  grid-row-gap: 15px;
`;

const Grid = ({ children }) => {
  return (
    <GridContainer>
      {children}
    </GridContainer>
  );
};

Grid.propTypes = {
  children: node,
};

Grid.defaultProps = {
  children: {},
};

export default Grid;
