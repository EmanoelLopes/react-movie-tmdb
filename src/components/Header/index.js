import React from 'react';
import { Link } from 'react-router-dom';
import Logo from 'components/Logo';
import tmdbLogo from 'assets/images/tmdb_logo.png';

function Header () {
  return (
    <header className="rm-header">
      <div className="container">
        <Link to="/">
          <span>
            <Logo />
            React Movie DB
          </span>
        </Link>
        <a href="https://www.themoviedb.org/?language=pt-BR" target="_blank" rel="noopener noreferrer">
          <img src={tmdbLogo} alt="The Movie DB" />
        </a>
      </div>
    </header>
  );
}

export default Header;
