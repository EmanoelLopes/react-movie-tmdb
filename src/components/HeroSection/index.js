import React from 'react';
import styled from 'styled-components';
import { string } from 'prop-types';

const SECTION = styled.section`
  display: flex;
  flex-direction: column;
`;

const HeroSection = ({ title, description }) => (
  <SECTION>
    <h1>{title}</h1>
    <p>{description}</p>
  </SECTION>
);

HeroSection.propTypes = {
  title: string.isRequired,
  description: string.isRequired,
};

export default HeroSection;
