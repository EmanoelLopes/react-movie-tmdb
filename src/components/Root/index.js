import React from 'react';
import { node } from 'prop-types';
import { Provider } from 'react-redux';
import store from 'store';

const Root = ({ children }) => (
  <Provider store={store}>
    {children}
  </Provider>
);

Root.propTypes = {
  children: node,
};

Root.defaultProps = {
  children: {},
};

export default Root;
