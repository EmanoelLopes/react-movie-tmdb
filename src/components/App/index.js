import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import Routes from 'routes';
import { GlobalStyle } from 'utils/styled';
import Header from 'components/Header';
import Main from 'components/Main';
import Footer from 'components/Footer';
import locale from 'lang';

locale.init();

function App() {
  return (
    <React.Fragment>
      <GlobalStyle />
      <BrowserRouter>
        <React.Fragment>
          <Header />
          <Main>
            <Routes />
          </Main>
          <Footer />
        </React.Fragment>
      </BrowserRouter>
    </React.Fragment>
  );
}

export { App };
