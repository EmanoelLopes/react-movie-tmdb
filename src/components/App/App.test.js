import React from 'react';
import { mount } from 'enzyme';
import Root from 'components/Root';
import { App } from 'components/App';

let wrapped;

beforeEach(() => {
  wrapped = mount(
    <Root>
      <App />
    </Root>
  );
});

afterEach(() => {
  wrapped.unmount();
});

describe('<App />', () => {
  it('should match snapshot', () => {
    expect(wrapped).toMatchSnapshot();
  });
});
