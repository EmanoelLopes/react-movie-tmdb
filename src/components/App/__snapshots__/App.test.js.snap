// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<Root>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <App>
      <GlobalStyleComponent />
      <BrowserRouter>
        <Router
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
        >
          <Header>
            <header>
              React Movie DB
            </header>
          </Header>
          <Main>
            <main>
              <Routes>
                <Switch>
                  <Route
                    component={[Function]}
                    computedMatch={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/",
                        "url": "/",
                      }
                    }
                    exact={true}
                    location={
                      Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      }
                    }
                    path="/"
                  >
                    <Home
                      history={
                        Object {
                          "action": "POP",
                          "block": [Function],
                          "createHref": [Function],
                          "go": [Function],
                          "goBack": [Function],
                          "goForward": [Function],
                          "length": 1,
                          "listen": [Function],
                          "location": Object {
                            "hash": "",
                            "pathname": "/",
                            "search": "",
                            "state": undefined,
                          },
                          "push": [Function],
                          "replace": [Function],
                        }
                      }
                      location={
                        Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        }
                      }
                      match={
                        Object {
                          "isExact": true,
                          "params": Object {},
                          "path": "/",
                          "url": "/",
                        }
                      }
                    >
                      <h1>
                        Home
                      </h1>
                    </Home>
                  </Route>
                </Switch>
              </Routes>
            </main>
          </Main>
          <Footer>
            <footer>
               Copyright Â© 2019 
            </footer>
          </Footer>
        </Router>
      </BrowserRouter>
    </App>
  </Provider>
</Root>
`;
